src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1439,6 +1439,18 @@ class ArnoldRenderer : public IECoreScenePreview::Renderer
 				AiNodeSetStr( options, "shader_searchpath", s.c_str() );
 				return;
 			}
+			else if( boost::starts_with( name.c_str(), "ai:declare:" ) )
+			{
+				const AtParamEntry *parameter = AiNodeEntryLookUpParameter( AiNodeGetNodeEntry( options ), name.c_str() + 11 );
+				if( parameter )
+				{
+					IECore::msg( IECore::Msg::Warning, "IECoreArnold::Renderer::option", boost::format( "Unable to declare existing option \"%s\"." ) % (name.c_str() + 11) );
+				}
+				else
+				{
+					ParameterAlgo::setParameter( options, name.c_str() + 11, value );
```

Comment:
Bear in mind that the value may be NULL - this is used in interactive rendering to remove a previous option that is no longer present. So I think you need the same here as there is in the "ai:" case below - using `AiNodeResetParameter()` when value is NULL. You might want to fire up an IPR session to check it works though (don't worry about a unit test for this bit, it'd be a real pain to set up).